gnosis/safe/safe.py

```diff
@@ -519,16 +522,28 @@ def contract(self) -> Contract:
 
     @cached_property
     def domain_separator(self):
-        return fast_keccak(
-            encode_abi(
-                ["bytes32", "uint256", "address"],
-                [
-                    self.DOMAIN_TYPEHASH,
-                    self.ethereum_client.get_chain_id(),
-                    self.address,
-                ],
+        version = self.contract.functions.VERSION().call()
+        if version == "1.3.0":
+            return fast_keccak(
+                encode_abi(
+                    ["bytes32", "uint256", "address"],
+                    [
+                        self.DOMAIN_TYPEHASH_V1_3_0,
+                        self.ethereum_client.get_chain_id(),
+                        self.address,
+                    ],
+                )
+            )
+        else:
```

Comment:
I think we should check the version here again and throw an error if no version was matched.

Else we are implicitly falling back to version 1 and ignoring a possible error.

Reply:
Done https://github.com/safe-global/safe-eth-py/pull/422/commits/f1ac04392efe6c31df0dc636a931ade86c92155e 
I added an Exception to indicate a version error.